solr/core/src/java/org/apache/solr/update/UpdateLog.java

```diff
@@ -2097,10 +2098,19 @@ public void doReplay(TransactionLog translog) {
         UpdateRequestProcessorChain processorChain = req.getCore().getUpdateProcessingChain(null);
         Collection<UpdateRequestProcessor> procPool =
             Collections.synchronizedList(new ArrayList<>());
+        final var params =
+            new MapSolrParams(
+                Map.of(
+                    DISTRIB_UPDATE_PARAM,
+                    FROMLEADER.toString(),
+                    DistributedUpdateProcessor.LOG_REPLAY,
+                    "true"));
         ThreadLocal<UpdateRequestProcessor> procThreadLocal =
             ThreadLocal.withInitial(
                 () -> {
-                  var proc = processorChain.createProcessor(req, rsp);
+                  // SolrQueryRequest is not thread-safe, so use a copy when creating URPs
+                  final var solrQueryRequest = new LocalSolrQueryRequest(uhandler.core, params);
```

Comment:
I'd call this var localRequest -- "local" being an important part.

Reply:
Done